### PR TITLE
Throw an error if a form factor or LCDA factory fails to make its target

### LINKED ISSUE #616: Throw an error if a form factor or LCDA factory fails to make its target.
Should we make an explicit error if the key is not found?

_Originally posted by @mreboud in https://github.com/eos/eos/pull/614#discussion_r1136697814_

### PR DESCRIPTION
**Pull request by dvandyk**
Resolves #616.

### CODE REVIEW

File: eos/b-decays/b-to-vec-nu-nu.cc [hunk 1]
```diff
@@ -217,7 +213,7 @@ namespace eos
     Implementation<BToVectorDineutrino>::options
     {
         Model::option_specification(),
-        FormFactorFactory<PToP>::option_specification(),
+        FormFactorFactory<PToV>::option_specification(),
```
Comment (mreboud) [inline eos/b-decays/b-to-vec-nu-nu.cc:216]: Good catch, thanks :)

File: eos/form-factors/b-lcdas.cc [hunk 2]
```diff
@@ -43,12 +45,10 @@ namespace eos
...
         {
             result.reset( i->second(parameters, options) );
-        }
-        else
-        {
-            throw InternalError("Unknown value for 'name' for BMesonLCDAs::make");
+            return result;
         }
 
+        throw InternalError("Unknown B-meson LCDA model: " + name);
```
Comment (mreboud) [inline eos/form-factors/b-lcdas.cc:51]: Were you able to check that?

File: eos/form-factors/mesonic.cc [hunk 3]
```diff
@@ -485,14 +510,18 @@ namespace eos
     std::shared_ptr<FormFactors<VToV>>
     FormFactorFactory<VToV>::create(const QualifiedName & name, const Parameters & parameters, const Options & options)
     {
+        Context ctx("When creating a V->P form factor");
```
Comment (mreboud) [inline eos/form-factors/mesonic.cc:513]: This should be V -> V I think

**Review by mreboud — CHANGES_REQUESTED** [review]
I am mostly happy with the changes. There is one typo, and I would like to see a test for this :)

File: eos/form-factors/mesonic.cc (hunk 3 above)
Comment (dvandyk) [reply eos/form-factors/mesonic.cc:513]: Argh. I noticed it in one place, but overlooked the context. Fixing it right now.

File: eos/form-factors/b-lcdas.cc (hunk 2 above)
Comment (dvandyk) [reply eos/form-factors/b-lcdas.cc:51]: After revising: yes. All form factors and LCDAs have been instrumented with a test case ensuring that an exception is raised for nonsense inputs.

Comment (mreboud) [reply eos/form-factors/b-lcdas.cc:51]: Excellent, thanks a lot!

**Pull request merged**